app/src/androidTest/java/com/duckduckgo/app/cta/ui/CtaViewModelTest.kt

```diff
@@ -398,14 +397,42 @@ class CtaViewModelTest {
     }
 
     @Test
-    fun whenRefreshCtaOnWhileBrowsingAndConceptTestFeatureActiveThenReturnSerpCta() = runBlockingTest {
+    fun whenRefreshCtaOnSerpWhileBrowsingAndConceptTestFeatureActiveThenReturnSerpCta() = runBlockingTest {
         setConceptTestFeature()
         val site = site(url = "http://www.duckduckgo.com")
         val value = testee.refreshCta(coroutineRule.testDispatcher, isBrowserShowing = true, site = site)
 
         assertTrue(value is DaxDialogCta.DaxSerpCta)
     }
 
+    @Test
+    fun whenRefreshCtaOnSerpWhereSerpAndAnyNonSerpCtaShownAndWidgetCtaActiveThenReturnWidgetCta() = runBlockingTest {
+        setConceptTestFeature(SearchWidgetDaxCta)
+        givenSerpCtaShown()
+        givenAtLeastOneNonSerpCtaShown()
+        whenever(mockWidgetCapabilities.supportsStandardWidgetAdd).thenReturn(true)
+        whenever(mockWidgetCapabilities.hasInstalledWidgets).thenReturn(false)
+        whenever(mockDismissedCtaDao.exists(CtaId.DAX_DIALOG_SEARCH_WIDGET)).thenReturn(false)
+        val site = site(url = "http://www.duckduckgo.com")
+
+        val value = testee.refreshCta(coroutineRule.testDispatcher, isBrowserShowing = true, site = site)
+
+        assertTrue(value is DaxDialogCta.SearchWidgetCta)
+    }
+
+    @Test
+    fun whenRefreshCtaOnSerpSiteWhereSerpCtaShownAndWidgetCtaShownAndConceptTestActiveThenReturnNull() = runBlockingTest {
+        setConceptTestFeature(SearchWidgetDaxCta)
+        givenSerpCtaShown()
+        givenAtLeastOneNonSerpCtaShown()
+        whenever(mockDismissedCtaDao.exists(CtaId.DAX_DIALOG_SEARCH_WIDGET)).thenReturn(true)
+        val site = site(url = "http://www.duckduckgo.com")
+
+        val value = testee.refreshCta(coroutineRule.testDispatcher, isBrowserShowing = true, site = site)
```

Comment:
I'd rather we don't test if something is null as result of a business logic function

Reply:
Will this change only apply if `DaxDialogCta.NotNeeded` exists or also for `null` values in general?

Reply:
When a null value is a possible and expected value (whether it should be or not is another discussion) we should test it just like we would any other value. What's your reason for not wanting to test it @malmstein?